buildkite/bazel-central-registry/bcr_presubmit.py

```diff
@@ -257,6 +262,89 @@ def run_test(repo_location, task_config_file, task):
         return 1
 
 
+def validate_existing_modules_are_not_modified():
+    # Get all files that are Modified, Renamed, or Deleted.
+    bazelci.print_expanded_group("Checking if existing modules are not modified")
+    output = subprocess.check_output(
+        ["git", "diff", "main...HEAD", "--diff-filter=MRD", "--name-only", "--pretty=format:"]
+    )
+
+    # Check if any of the source.json, MODULE.bazel, or patch files are changed for an existing module.
+    NO_CHANGE_FILE_PATTERNS = [
+        r"modules\/([^\/]+)\/([^\/]+)\/source.json",
+        r"modules\/([^\/]+)\/([^\/]+)\/MODULE.bazel",
+        r"modules\/([^\/]+)\/([^\/]+)\/patches",
+    ]
+    changed_modules = []
+    for line in output.decode("utf-8").split():
+        for p in NO_CHANGE_FILE_PATTERNS:
+            s = re.match(p, line)
+            if s:
+                bazelci.eprint(line, "was changed")
+                changed_modules.append(s.groups())
+    if changed_modules:
+        error("Existing modules should not be changed:\n" + "\n".join([f"{name}@{version}" for name,version in changed_modules]))
+    else:
+        bazelci.eprint("No existing module was changed.")
+
+
+def validate_files_outside_of_modules_dir_are_not_modififed(modules):
+    # If no modules are changed at the same time, then we don't need to perform this check.
+    if not modules:
+        return
+    bazelci.print_expanded_group("Checking if any file changes outside of modules/")
+    output = subprocess.check_output(
+        ["git", "diff", "main...HEAD", "--name-only", "--pretty=format:", ":!modules"]
+    ).decode("utf-8").strip()
+    if output:
+        error("The following files should not be changed when adding a new module version:\n" + output)
+    else:
+        bazelci.eprint("Nothing changed outside of modules/")
+
+
+def should_bcr_validation_block_presubmit(modules):
+    if not modules:
+        return
+    bazelci.print_expanded_group("Running BCR validations:")
+    returncode = subprocess.run(
+        ["python3", "./tools/bcr_validation.py"] + [f"--check={name}@{version}" for name, version in modules]
+    ).returncode
+    # When a BCR maintainer view is required, the script should return 42.
+    if returncode == 42:
+        return True
+    if returncode != 0:
+        raise BcrPipelineException("BCR validation failed!")
```

Comment:
I'm assuming the error messages output by bcr_validation.py are also visible here?